### PR TITLE
feat: implement recently-added tree types to tree generator

### PR DESCRIPTION
**Pull request by dordsor21**
I believe this is what the trees should be like

image

### CODE REVIEW

File: worldedit-core/src/main/java/com/sk89q/worldedit/util/TreeGenerator.java [hunk 1]
```diff
@@ -108,6 +108,9 @@ public boolean generate(EditSession editSession, BlockVector3 pos) throws MaxCha
                 return editSession.getWorld().generateTree(this, editSession, pos.subtract(0, 1, 0));
             }
         },
+        MANGROVE("Mangrove tree", "mangrove"),
```
Comment (me4502) [inline worldedit-core/src/main/java/com/sk89q/worldedit/util/TreeGenerator.java:111]: This doesn't map these anywhere, so these won't do anything. They'll need to be mapped to tree features in the platforms

Comment (dordsor21) [reply worldedit-core/src/main/java/com/sk89q/worldedit/util/TreeGenerator.java:111]: Ah yeah mb, I was copying over from FAWE changes and we lazily just look up the enums via Bukkit's methods

**Comment by me4502** [comment]
Looks like tests are failing

**Comment by dordsor21** [comment]
Yeah, that'll be because of the bukkit module using 1.17 API I suppose, so the TreeType enum doesn't include cherry or mangrove trees. Not sure if/when you guys would want to switch to a later paper API?

**Comment by dordsor21** [comment]
👍 Won't be able to do any test until Thurs or Fri

**Comment by me4502** [comment]
Thanks, just bumped it so should pass now

**Pull request merged**